src/components/BusinessDonationHistory/BusinessDonationHistory.jsx

```diff
@@ -11,17 +25,45 @@ const BusinessDonationHistory = () => {
   const [selectedDonationId, setSelectedDonationId] = useState(null);
   const TABLE_HEADERS = ['Submitted By', 'Food Bank', 'Date', 'Action'];
 
+  // for pagination
+  const [currentTotalDonationNum, setCurrentTotalDonationNum] = useState(0);
+  const [currentPageNum, setCurrentPageNum] = useState(1);
+  const [pageLimit, setPageLimit] = useState(1);
+
+  // for search
+  const [searchInput, setSearchInput] = useState('');
+
+  // TEMPORARY DISABLE LINE SO THAT IN THE FUTURE IF YOU WANT TO ADD
+  // A DROP DOWN TO CHANGE PAGINATION NUMBER JUST USE THE SET FUNCTION
+  // AND REMOVE THE COMMENT
+  // eslint-disable-next-line no-unused-vars
+  const [paginationNumber, setPaginationNumber] = useState(13);
+
   useEffect(() => {
     const getData = async () => {
       try {
-        const response = await backend.get('/donation');
+        const response =
+          searchInput === ''
+            ? await backend.get(
+                `/donation/?donationsLimit=${paginationNumber}&pageNum=${currentPageNum}`,
+              )
```

Comment:
If statement isnt needed, if the search input is empty, the 2 queries would return the same thing regardless. SQL string matching with empty strings works the same way (those 4 lines can be deleted)

Reply:
Also edit the backend PR to remove the unecessary route (in the first query)